both/collections/competition.js

```diff
@@ -46,13 +49,19 @@ Competitions.attachSchema({
     max: 23.5*60,
     defaultValue: 23.5*60,
     custom: function() {
-      // require the competition calendar end time to be after the start time
-      var calendarStartMinutes = this.field("calendarStartMinutes").value;
-      var calendarEndMinutes = this.value;
-      if(calendarEndMinutes <= calendarStartMinutes) {
+      var obj = validationObject(this, ['calendarStartMinutes', 'calendarEndMinutes']);
+
+      if(obj.calendarEndMinutes <= obj.calendarStartMinutes) {
         return "calendarEndIsNotBeforeStart";
       }
-      return null;
+
+      var error = null;
+      ScheduleEvents.find({competitionId: this.docId}).forEach(function(event) {
+        if(event.endMinutes() > obj.calendarEndMinutes) {
+          error = "laterExistingEvents";
```

Comment:
Ah, damn forEach being a function! (you could also change it to a proper for loop, but lets not)

Your call how to handle this. Let it be, or change it ot the more functional style you have, although I think using underscore's [`some()` or `any()`](http://underscorejs.org/#some) would probably be more semantically correct here.